p2p/src/network/pubsub/p2p_network_pubsub_actions.rs

```diff
@@ -91,32 +101,65 @@ pub enum P2pNetworkPubsubAction {
 
     /// An error occured during the signing process.
     #[action_event(level = warn, fields(display(author), display(topic)))]
-    SignError { author: PeerId, topic: String },
+    SignError {
+        author: PeerId,
+        topic: String,
+    },
 
     /// Finalize the broadcasting of a signed message by attaching the signature.
     ///
     /// **Fields:**
     /// - `signature`: The cryptographic signature of the message.
-    BroadcastSigned { signature: Data },
+    BroadcastSigned {
+        signature: Data,
+    },
 
     /// Prepare an outgoing message to send to a specific peer.
-    OutgoingMessage { peer_id: PeerId },
+    OutgoingMessage {
+        peer_id: PeerId,
+    },
 
     /// Clear the outgoing message state for a specific peer after sending.
-    OutgoingMessageClear { peer_id: PeerId },
+    OutgoingMessageClear {
+        peer_id: PeerId,
+    },
 
     /// An error occured during the sending of an outgoing message.
     ///
     /// **Fields:**
     /// - `msg`: The protobuf message that failed to send.
     #[action_event(level = warn, fields(display(peer_id), debug(msg)))]
-    OutgoingMessageError { msg: pb::Rpc, peer_id: PeerId },
+    OutgoingMessageError {
+        msg: pb::Rpc,
+        peer_id: PeerId,
+    },
 
     /// Send encoded data over an outgoing stream to a specific peer.
     ///
     /// **Fields:**
     /// - `data`: The encoded data to be sent.
-    OutgoingData { data: Data, peer_id: PeerId },
+    OutgoingData {
+        data: Data,
+        peer_id: PeerId,
+    },
+
+    BroadcastValidationCallback {
+        message_id: P2pNetworkPubsubMessageCacheId,
+    },
+
+    BroadcastValidatedMessage {
```

Comment:
This should also be after `BroadcastValidatedMessage`, it doesn't happen before, it happens after